### PR TITLE
Processing data model

### PR DESCRIPTION
**Pull request by rbd3**
### Change made:
👉 Create model classes for all entities
👉 Set up associations between models.

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -0,0 +1,5 @@
+class User < ApplicationRecord
+  has_many :posts, foreign_key: 'author_id'
+  has_many :likes
+  has_many :comments
```
Comment (ShadyShawkat) [inline app/models/user.rb]: - [ ] Kindly ensure that the foreign key is referenced correctly in the comments and likes, much like what you did with the posts: `foreign_key: 'author_id'`.
- [ ] Please create a method for the user model that returns the 3 most recent posts for a given user, as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20returns%20the%203%20most%20recent%20posts%20for%20a%20given%20user.) requirement.

File: app/models/like.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Like < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
```
Comment (ShadyShawkat) [inline app/models/like.rb]: - [ ] Kindly user `author` here instead of `user`.

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :likes
+  has_many :comments
```
Comment (ShadyShawkat) [inline app/models/post.rb:4]: - [ ] Please create two methods for the post model which are given in the following [requirements](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20updates,for%20a%20given%20post.).

File: app/models/comment.rb [hunk 4]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :author, class_name: 'User', foreign_key: 'author_id'
+  belongs_to :post
```
Comment (ShadyShawkat) [inline app/models/comment.rb:3]: - [ ] Please create a method for the comment model which updates the comments counter for a post,as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20updates%20the%20comments%20counter%20for%20a%20post.) requirement.

File: app/models/like.rb (hunk 2 above)
Comment (ShadyShawkat) [inline app/models/like.rb]: - [ ] Please create a method for the like model which updates the comments counter for a post,as per [this](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/projects/Processing_data_model_project_v1_1.md#:~:text=A%20method%20that%20updates%20the%20likes%20counter%20for%20a%20post.) requirement.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi @rbd3, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by binhussen — APPROVED** [review]
Hi @rbd3 , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉

## Highlights
- Proceeded data models ✔️
- Well Stractured Code ✔️
- Used Git flow ✔️
## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**